src/common/shared_cache.hpp

```diff
@@ -62,6 +65,14 @@ class SharedLRU {
     }
   }
 
+  VPtr lru_lookup(const K& key) {
+    if (const auto i = contents.find(key); i != std::end(contents)) {
+      lru.splice(std::begin(lru), lru, i->second);
+      return lru.front().second;
+    }
+    return VPtr();
+  }
+
```

Comment:
@rzarzynski . when add new element which beyond size of lru. it will remove from contents.  If the weak_refs still  contain, for lookup function, it will search contents and weak_refs.  I'm not sure the probability for this case.

Reply:
@majianpeng after a second thought, would it be better if we just drop the `weak_refs`? as in a single-threaded environment,  we won't have a pointer **being** deleted, and hence need to wait for it. if that's the case, i am all for forking `shared_cache.hpp` under `crimson/` directory. as it will divert the code even more.  sorry for the bikeshedding.

Reply:
@tchaikov . sorry later to response.  the function of weak_refs is make wrong-refs leave lru because size. And wrong-refs still use and don't free. So the next find will readd wrong-refs to lru by weak_refs.   For single-threaded still has this case.